packages/bundler/src/bundler.ts

```diff
@@ -121,7 +128,14 @@ export class Bundler implements Subscribable<BundlerMessage, undefined> {
 
       this.channel.send({ type: 'UPDATE' });
     } catch(error) {
-      this.channel.send({ type: 'ERROR', error });
+      let message = error.message
+      if(error.loc) {
+        message += `\n\n${error.loc.file}:${error.loc.line}:${error.loc.column}`;
```

Comment:
I think you can get a better error message out of this and maybe this is not the right place to be doing this.

My take on this would be to put as much in the state and let whatever is being notified of the state change construct the error message.

I did [this](https://github.com/thefrontside/bigtest/blob/reporter/packages/reporter/src/out/console/console.ts#L129) on a spike branch which puts a nice error out to the console.

Reply:
a string message in the state gives less wriggle room.

Reply:
Hmm, interesting. We might be able to render this error a littler nicer, on the other hand we make ourselves dependent on the format of an error generated in a completely different package. I'm a bit torn on this.

@cowboyd which option do you think is better?

Reply:
I think that error structure of `filename`, `linenumber` and `column` is pretty common even though it is coming from rollup in this instance

This [package](https://babeljs.io/docs/en/babel-code-frame) will print this nicely to the UI if you have the filename, line number and column.

It is a babel package so knows nothing about rollup.  Rollup has kindly retrieved this info for us.

Reply:
My first inclination would be to do both. If we can provide it in a canonical string format, but also provide access to the underlying structure then we can have the best of both worlds. UI's that don't want to care have a quick fix, but those that want to take on the coupling to a richer data type are allowed to. 

If we could, for example, access the underlying errors emitted by the TS compiler, they seem like very detailed and unlikely to change. http://definitelytyped.org/docs/typescript-services--typescriptServices/classes/typescript.diagnostic.html Those seem really stable, and unlike rollup and babel, coupled to the right 3rd party library, TypeScript itself.

Alternatively, we could roll our own types for referencing source locations. That would give us the most control, but it would also mean that it was the most work for us :)

Reply:
> My first inclination would be to do both.

What a great answer!

There might be some benefit in having a generic error type and possibly generic stackframe.

The codeframe comes from a stackframe.

If you want a codeframe to display, then it will need a filename, line number and column.  That is pretty ubiquitous and unlikely to change but are called slightly different things everywhere.